src/acceptor/builder.rs

```diff
@@ -0,0 +1,97 @@
+use std::sync::Arc;
+
+use hyper::server::conn::AddrIncoming;
+use rustls::ServerConfig;
+
+use super::TlsAcceptor;
+/// Builder for [`TlsAcceptor`]
+pub struct AcceptorBuilder<State>(State);
+
+/// State of a builder that needs a TLS client config next
+pub struct WantsTlsConfig(());
+
+impl AcceptorBuilder<WantsTlsConfig> {
+    /// Creates a new [`AcceptorBuilder`]
+    pub fn new() -> Self {
+        Self(WantsTlsConfig(()))
+    }
```

Comment:
Let's make this private for now, so that the only way to construct it is through `Acceptor::builder()`.